src/indexBy.ts

```diff
@@ -2,16 +2,22 @@ import { purry } from './purry';
 import { PredIndexedOptional, PredIndexed } from './_types';
 
 /**
- * Converts a list of objects into an object indexing the objects by the given key.
+ * Converts a list of objects into an object indexing the objects by the given key
+ * (casted to a string). Use the strict version to maintain the given key's type, so
+ * long as it is a valid `PropertyKey`.
+ *
```

Comment:
Keep the docs in one line; our docs site would add the word breaks as needed.
```suggestion
 * Converts a list of objects into an object indexing the objects by the given key (casted to a string).
 * Use the strict version to maintain the given key's type, so long as it is a valid `PropertyKey`.
 *
```